clarity/contracts/arkadiko-liquidation-rewards-diko-v1-1.clar

```diff
@@ -0,0 +1,101 @@
+(use-trait liquidation-rewards-trait .arkadiko-liquidation-rewards-trait-v1.liquidation-rewards-trait)
+
+;; Errors
+(define-constant ERR-NOT-AUTHORIZED u34401)
+(define-constant ERR-EPOCH-NOT-ENDED u34002)
+(define-constant ERR-EMERGENCY-SHUTDOWN-ACTIVATED u34003)
+
+;; Variables
+;; TODO - SET FOR MAINNET
+;; TODO - UPDATE REWARDS FOR STAKING POOLS
+(define-data-var end-epoch-block uint (+ block-height u1440)) ;; 10 days after deploy
+(define-data-var epoch-rate uint u100000) ;; 10%
+(define-data-var blocks-per-epoch uint u720) ;; 5 days
+(define-data-var shutdown-activated bool false)
+
+(define-read-only (get-end-epoch-block)
+  (ok (var-get end-epoch-block))
+)
+
+(define-read-only (get-epoch-rate)
+  (ok (var-get epoch-rate))
+)
+
+(define-read-only (get-blocks-per-epoch)
+  (ok (var-get blocks-per-epoch))
+)
+
+(define-read-only (get-epoch-info)
+  (ok {
+    blocks: (var-get blocks-per-epoch),
+    rate: (var-get epoch-rate),
+    end-block: (var-get end-epoch-block)
+  })
+)
+
+(define-read-only (get-shutdown-activated)
+  (or
+    (unwrap-panic (contract-call? .arkadiko-dao get-emergency-shutdown-activated))
+    (var-get shutdown-activated)
+  )
+)
+
+;; @desc toggles the killswitch
+(define-public (toggle-shutdown)
+  (begin
+    (asserts! (is-eq tx-sender (contract-call? .arkadiko-dao get-guardian-address)) (err ERR-NOT-AUTHORIZED))
+
+    (ok (var-set shutdown-activated (not (var-get shutdown-activated))))
+  )
+)
+
+;; ---------------------------------------------------------
+;; Add rewards
+;; ---------------------------------------------------------
+
+;; @desc Amount of rewards that can be added
+(define-read-only (get-rewards-to-add)
+  (let (
+    (total-block-rewards (contract-call? .arkadiko-diko-guardian-v1-1 get-staking-rewards-per-block))
+    (rewards-to-add (/ (* total-block-rewards (var-get blocks-per-epoch) (var-get epoch-rate)) u1000000))
```

Comment:
Q: is this actually correct? Doesn't the reward go down slightly each block? Or should we ignore that as a rounding number?

Reply:
It is slightly off indeed. To calculate the exact rewards we would need to loop over every block which is not feasible. The approximation should be fine if the periods are not too long. We have the same approach for staking rewards.